src/gamma_inc.jl

```diff
@@ -1014,8 +1014,22 @@ function _gamma_inc_inv(a::Float64, p::Float64, q::Float64)
     return x
 end
 
-_gamma_inc_inv(a::Float32, p::Float32, q::Float32) = Float32(_gamma_inc_inv(Float64(a), Float64(p), Float64(q)))
-_gamma_inc_inv(a::Float16, p::Float16, q::Float16) = Float16(_gamma_inc_inv(Float64(a), Float64(p), Float64(q)))
+function _gamma_inc_inv(a::T, p::T, q::T) where {T <: Union{Float16, Float32}}
+    p64, q64 = if p < q
+        _q = 1 - Float64(p)
+        if q != T(_q)
+            throw(ArgumentError("p + q must equal one but was $(p + q)"))
+        end
+        Float64(p), _q
+    else
+        _p = 1 - Float64(q)
+        if p != T(_p)
+            throw(ArgumentError("p + q must equal one but was $(p + q)"))
+        end
+        _p, Float64(q)
+    end
```

Comment:
Why not just
```jl
p + q == one(T) || throw(ArgumentError("p + q must equal one but was $(p + q)"))
p64, q64 = p < q ? (Float64(q), 1 - Float64(p)) : (1 - Float64(q), Float64(q))
```

Reply:
Yeah. That avoids some replication. I'll change it to something along these lines. However, I find it easier to read with the branches on separate lines and it also makes it easier to verify coverage.